File: examples/sample-rp/src/main/java/com/crosscert/passkey/samplerp/web/PageController.java
package com.crosscert.passkey.samplerp.web;

import com.crosscert.passkey.samplerp.session.SessionKeys;
import com.crosscert.passkey.samplerp.user.SampleRpUser;
import jakarta.servlet.http.HttpSession;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class PageController {

    @GetMapping("/")
    public String index(HttpSession s, Model m) {
        SampleRpUser u = (SampleRpUser) s.getAttribute(SessionKeys.USER);
        m.addAttribute("user", u);
        return "index";
    }

    @GetMapping("/register")
    public String register() { return "register"; }

    @GetMapping("/login")
    public String login() { return "login"; }
}

File: rp-app/src/main/kotlin/com/crosscert/passkey/rpapp/web/PageController.kt
package com.crosscert.passkey.rpapp.web

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping

@Controller
class PageController {

    @GetMapping("/")
    fun index(): String = "index"

    @GetMapping("/register")
    fun register(): String = "register"

    @GetMapping("/login")
    fun login(): String = "login"
}
